File: app/src/main/java/com/wjf/self_demo/adapter/IndexListAdapter.java
package com.wjf.self_demo.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.View;

import com.wjf.self_demo.R;
import com.wjf.self_demo.databinding.ItemListIndexBinding;
import com.wjf.self_demo.entity.IndexListMenu;
import com.wjf.self_library.common.CommonAdapter;

import java.util.List;

/**
 * @author : Wangjf
 * @date : 2021/4/6
 */
public class IndexListAdapter extends CommonAdapter<ItemListIndexBinding, IndexListMenu> {


    @Override
    protected int setLayout() {
        return R.layout.item_list_index;
    }

    @Override
    protected void setHolder(IndexListMenu entity, ItemListIndexBinding view) {
        view.menuText.setText(entity.getDes());
        view.menuText.setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Intent intent = new Intent();
                        intent.putExtra("data", entity.getData());
                        intent.setClass(context, entity.getGotoClass());
                        context.startActivity(intent);
                    }
                });
    }
}

File: app/src/main/java/com/wjf/self_demo/adapter/IndexListAdapter.kt
package com.wjf.self_demo.adapter

import android.content.Intent
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.SelfViewHolder
import com.wjf.self_demo.databinding.ItemListIndexBinding
import com.wjf.self_demo.entity.IndexListMenu
import org.jxxy.debug.corekit.recyclerview.SingleTypeAdapter
import org.jxxy.debug.corekit.util.singleClick

/**
 * @author : Wangjf
 * @date : 2021/4/6
 */
class IndexListAdapter : SingleTypeAdapter<IndexListMenu>() {
    private var total = 0L

    class IndexListViewHolder(view: ItemListIndexBinding) : SelfViewHolder<ItemListIndexBinding, IndexListMenu>(view) {
        override fun setHolder(entity: IndexListMenu) {
            val start = System.currentTimeMillis()
            view.menuText.text = entity.des
            view.root.singleClick { v: View? ->
                val intent = Intent()
                intent.putExtra("data", entity.data)
                intent.setClass(context, entity.gotoClass)
                context.startActivity(intent)
            }
            Log.d("wjftc", "onBindViewHolder: ${System.currentTimeMillis() - start} ms")
        }
    }

    override fun createViewHolder(viewType: Int, inflater: LayoutInflater, parent: ViewGroup): RecyclerView.ViewHolder {
        val start = System.currentTimeMillis()
        val holder = IndexListViewHolder(ItemListIndexBinding.inflate(inflater, parent, false))
        val dif = System.currentTimeMillis() - start
        total += dif
        Log.d("wjftc", "createViewHolder: $dif ms in ${Thread.currentThread().name} $total ms")
        return holder
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)
    }

    fun addAll(list: List<IndexListMenu>?) {
        if (list.isNullOrEmpty()) {
            return
        }
        data.addAll(list)
        notifyDataSetChanged()
    }
}
